Source/CarthageKit/Xcode.swift

```diff
@@ -1395,59 +1402,77 @@ public func architecturesInFramework(frameworkURL: NSURL) -> SignalProducer<Stri
 	return SignalProducer.attempt { () -> Result<NSURL, CarthageError> in
 			return binaryURL(frameworkURL)
 		}
-		.flatMap(.Merge) { binaryURL -> SignalProducer<String, CarthageError> in
-			let lipoTask = Task("/usr/bin/xcrun", arguments: [ "lipo", "-info", binaryURL.path!])
+		.flatMap(.Merge) { binaryURL in
+			return architecturesInBinary(binaryURL)
+		}
+}
 
-			return launchTask(lipoTask)
-				.ignoreTaskData()
-				.mapError(CarthageError.TaskError)
-				.map { NSString(data: $0, encoding: NSUTF8StringEncoding) ?? "" }
-				.flatMap(.Merge) { output -> SignalProducer<String, CarthageError> in
-					let characterSet = NSMutableCharacterSet.alphanumericCharacterSet()
-					characterSet.addCharactersInString(" _-")
-
-					let scanner = NSScanner(string: output as String)
-
-					if scanner.scanString("Architectures in the fat file:", intoString: nil) {
-						// The output of "lipo -info PathToBinary" for fat files
-						// looks roughly like so:
-						//
-						//     Architectures in the fat file: PathToBinary are: armv7 arm64
-						//
-						var architectures: NSString?
-
-						scanner.scanString(binaryURL.path!, intoString: nil)
-						scanner.scanString("are:", intoString: nil)
-						scanner.scanCharactersFromSet(characterSet, intoString: &architectures)
-
-						let components = architectures?
-							.componentsSeparatedByString(" ")
-							.filter { !$0.isEmpty }
-
-						if let components = components {
-							return SignalProducer(values: components)
-						}
-					}
+/// Returns a signal of all architectures present in a given dSYM.
+public func architecturesInDSYM(dSYMURL: NSURL) -> SignalProducer<String, CarthageError> {
+	if let DWARFName = dSYMURL.URLByDeletingPathExtension?.URLByDeletingPathExtension?.lastPathComponent {
+		let DWARFURL = dSYMURL.URLByAppendingPathComponent("Contents/Resources/DWARF/\(DWARFName)")
```

Comment:
Let's rename `DWARFName` to `binaryName` and `DWARFURL` to `binaryURL` for consistency with `architecturesInFramework()`.